NitroxServer/ServerAutoFacRegistrar.cs

```diff
@@ -29,7 +33,51 @@ public virtual void RegisterDependencies(ContainerBuilder containerBuilder)
 
         private static void RegisterCoreDependencies(ContainerBuilder containerBuilder)
         {
-            containerBuilder.Register(c => ServerConfig.Load()).SingleInstance();
+            containerBuilder.Register(c =>
+            {
+                string saveDir = null;
+                foreach (string arg in Environment.GetCommandLineArgs())
+                {
+                    if (arg.StartsWith(WorldManager.SavesFolderDir, StringComparison.OrdinalIgnoreCase) && Directory.Exists(arg))
+                    {
+                        saveDir = arg;
+                        break;
+                    }
+                }
+                if (saveDir == null)
+                {
+                    // Check if there are any save files
+                    IEnumerable<WorldManager.Listing> WorldList = WorldManager.GetSaves();
+                    if (WorldList.Any())
+                    {
+                        // Get last save file used
+                        string LastSaveAccessed = WorldList.ElementAtOrDefault(0).WorldSaveDir;
+                        if (WorldList.Count() > 1)
+                        {
+                            for (int i = 1; i < WorldList.Count(); i++)
+                            {
+                                if (File.GetLastWriteTime(Path.Combine(WorldList.ElementAtOrDefault(i).WorldSaveDir, "WorldData.json")) > File.GetLastWriteTime(LastSaveAccessed))
+                                {
+                                    LastSaveAccessed = WorldList.ElementAtOrDefault(i).WorldSaveDir;
+                                }
+                            }
+                        }
+                        saveDir = LastSaveAccessed;
+                    }
+                    else
+                    {
+                        // Create new save file
+                        saveDir = Path.Combine(WorldManager.SavesFolderDir, "My World");
+                        Directory.CreateDirectory(saveDir);
+                        ServerConfig serverConfig = ServerConfig.Load(saveDir);
+                        Log.Debug($"No save file was found, creating a new one...");
+                    }
```

Comment:
This shouldn't be here, this code belongs to our Dependency Injector Container which has nothing to do with the world managing and is prone to I/O errors that are not catched or prevented in this code

Reply:
Does this resolve this issue?: https://github.com/SubnauticaNitrox/Nitrox/pull/1808/commits/e74d32b2a3a90f28ef78c8b0c95a09ee0a70c873